xbmc/addons/addoninfo/AddonType.h

```diff
@@ -71,6 +71,12 @@ typedef enum
   ADDON_MAX
 } TYPE;
 
+const std::set<TYPE> ADDON_DEPENDENCY_TYPES = {
```

Comment:
This should be encapsulated in CAddonType. I don't see why it should be accessed from outside this class. It could be private and static.

Reply:
For me the same, we must prevent the use of such parts as much as possible from outside of addon folder.

To have static and private is better, also to use not a uppercase name (the `ADDON_DEPENDENCY_TYPES`), for me means them more a `#define` or a "C" structure style.

```cpp
const std::set<TYPE> CAddonType::addonDependencyTypes = {
```

Also to use data lists within headers for me not the best, the C++ optimization normally makes it clean but data is data :smirk:

Reply:
Also, maybe this should not be in caps. `AddonDependencyTypes` would be better.

Reply:
Thanks! Actually there are a lot of static functions (and sets) that are internal to `CAddonDirectory` that would be better moved to `AddonTypes` in the future. I struggled a bit to visualize how to best call `IsDependencyType` without exposing the method from that file and include the filesystem/addon header everywhere. Maybe we can keep this in mind to move some of the other type-related functions from filesystem to addons/Addontypes in the future